python/lsst/obs/base/gen2to3/convertTests.py

```diff
@@ -90,11 +90,23 @@ class ConvertGen2To3TestCase:
     instrument name and any refcats, if ``refcats`` is non-empty above.
     Typically the only additional one necessary would be "skymaps"."""
 
+    detectorKey = "ccd"
+    """Key to use in a gen2 dataId to refer to a detector."""
+
+    exposureKey = "visit"
+    """Key to use in a gen2 dataId to refer to a visit or exposure."""
+
     def setUp(self):
         self.gen3root = tempfile.mkdtemp()
         self.gen2Butler = lsst.daf.persistence.Butler(root=self.gen2root, calibRoot=self.gen2calib)
         # This command is in obs_base, and we use the one that has been setup and scons'ed.
         self.cmd = "convert_gen2_repo_to_gen3.py"
+
+        # if the collections set is empty we do not add to it, we create
+        # a new instance version. Without this each subclass would add
+        # to the same set.
+        if not self.collections:
+            self.collections = set()
```

Comment:
The detector key and exposure key are fine because they are string constants. I probably should change all the lists to read only empty tuples and maybe change collections to a frozenset.

Reply:
I tend to like the extra checks and clarity of `abstractmethod` better anyway, but I certainly agree that the non-constant values are the dangerous ones.

And on that note, I _think_ this problem is now breaking the `obs_decam` tests on the DM-21849 branch (https://ci.lsst.codes/blue/organizations/jenkins/stack-os-matrix/detail/stack-os-matrix/31380/tests).  If you're about ready to merge this ticket, I'm happy to just wait and rebase, or to take care of making these all immutable (or `abstractmethod`-based) on DM-21849 myself.  It just doesn't make sense for both of us to do that.

Reply:
I could merge pretty quickly. I think the curated calibrations discussion is not technically blocking this since I haven't yet moved the code to obs_base for that. I can take a look on Sunday at your other comments.